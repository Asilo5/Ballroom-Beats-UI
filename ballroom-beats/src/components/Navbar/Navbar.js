import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { createStackNavigator } from 'react-navigation-stack';
import { withNavigation } from 'react-navigation';

class Navbar extends Component {

  render() {
      return (
          <View style={styles.container}>
            <Text>Navbar</Text>
          </View>
      )      
  }
};  

const AppNavigator = createStackNavigator({
    Navbar: {
      screen: Navbar,
    },
});
  
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'white',
        alignItems: 'center',
        justifyContent: 'center',
    }
});

export default withNavigation(Navbar);