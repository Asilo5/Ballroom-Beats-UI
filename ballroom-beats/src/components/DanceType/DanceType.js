import React, { Component } from 'react';
import { View, Text, StyleSheet, Picker } from 'react-native';
import { createStackNavigator } from 'react-navigation-stack';
import { withNavigation } from 'react-navigation';
import Icon from 'react-native-vector-icons/AntDesign';
import Navbar from '../Navbar/Navbar';

class DanceType extends Component {
  constructor() {
    super()
    this.state= {
        PickerSelectedDance : ''
    }
  }
    // method for taking value, passing down to songPick for what songs to display
    getSelectedPickerDance = () =>{
    
    }
// Lets try not to use redux. Instead pass dance type to song pick to measure the difficulty of song that is available
  render() {
      return (
          <View style={styles.container}>
            <Text style={styles.dance}>Pick Your Dance Type</Text>
                <Picker
                    style={styles.picker}
                    itemStyle={styles.picker_text}
                    selectedValue={this.state.PickerSelectedDance}
                    onValueChange={(itemValue, itemIndex) => 
                        this.setState({PickerSelectedDance: itemValue})} >
                    <Picker.Item label="--Pick a Dance--" value="" />
                    <Picker.Item label="Beginner: Waltz" value="Waltz" />
                    <Picker.Item label="Intermediate: Bachata" value="Bachata" />
                    <Picker.Item label="Advanced: Swing" value="Swing" />
                </Picker>
            <Icon.Button 
                style={styles.button} 
                name="arrowright" 
                onPress={() => this.props.navigation.navigate('SongPick')} 
                title="NEXT">
                <Text style={styles.next}>Next</Text>
            </Icon.Button>
            <Navbar />
          </View>
      )      
  }
} 

const AppNavigator = createStackNavigator({
    DanceType: {
      screen: DanceType,
    },
});
  
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'black',
        alignItems: 'center',
        justifyContent: 'center',
    },
    dance: {
        fontSize: 40,
        color: 'white',
        marginBottom: 10,
    },
    picker: {
        width: 350,
        borderColor: 'lime',
        borderWidth: 1,
        marginBottom: 10,
        // backgroundColor: 'wheat',
        height: 64,
    },
    picker_text: {
        color: 'lime',
        fontWeight: 'bold',
        fontSize: 30,
        height: 64,
    },
    item: {
        color: 'lime',
        fontWeight: 'bold',
        fontSize: 30,
    },
    button: {
        backgroundColor: 'green',
        fontSize: 25,
    },
    next: {
        fontSize: 25,
    }
});

export default withNavigation(DanceType);