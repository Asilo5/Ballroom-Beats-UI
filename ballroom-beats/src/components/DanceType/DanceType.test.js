import React from 'react';
import renderer from 'react-test-renderer';
import DanceType from '../DanceType/DanceType';
import Navbar from '../Navbar/Navbar';

jest.mock('../Navbar/Navbar')

const mockProps = {
   navigation: {
     navigate: jest.fn(),
     getParam: jest.fn(),
   }
 };

describe('DanceType', () => {

    Navbar.mockImplementation(() => {
      return null
    })
   it('should match snapshot', () => {
    const dance = renderer.create(<DanceType navigation={mockProps.navigation} />).toJSON();
    expect(dance).toMatchSnapshot();
   });

});
