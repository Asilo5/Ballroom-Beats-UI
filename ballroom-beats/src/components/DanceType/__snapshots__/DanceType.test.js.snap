// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DanceType should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#666363",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 35,
        "fontWeight": "bold",
        "marginBottom": 10,
        "textShadowColor": "#000",
        "textShadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "textShadowRadius": 2,
      }
    }
  >
    Pick Your Dance Type
  </Text>
  <View
    style={
      Object {
        "borderColor": "#A9C344",
        "borderRadius": 3,
        "borderWidth": 1,
        "height": 66,
        "marginBottom": 10,
        "width": 350,
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "-- Pick a Dance --",
            "textColor": undefined,
            "value": "",
          },
          Object {
            "label": "Novice: Waltz",
            "textColor": undefined,
            "value": "Waltz",
          },
          Object {
            "label": "Pro: Bachata",
            "textColor": undefined,
            "value": "Bachata",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          Object {
            "backgroundColor": "#39373A",
            "color": "#A9C344",
            "fontSize": 30,
            "fontWeight": "bold",
            "height": 64,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#007AFF",
          "borderRadius": 5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 8,
          },
          Object {
            "backgroundColor": "#007AFF",
            "borderRadius": 5,
          },
          Array [
            Object {
              "backgroundColor": "#A9C344",
              "width": 130,
            },
            Object {
              "backgroundColor": "#3E3E3E",
            },
          ],
        ]
      }
      title="NEXT"
    >
      <Text />
      <Text
        style={
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;
